python/lsst/ap/verify/dataset.py

```diff
@@ -53,9 +53,8 @@ class Dataset(object):
     """
 
     def __init__(self, datasetId):
-        try:
-            datasetPackage = self._getDatasetInfo()[datasetId]
-        except KeyError:
+        datasetPackage = self._getDatasetInfo()[datasetId]
```

Comment:
You're reasonably confident that trying to get a non-existent dataset will return None and won't fail in some other way?

Reply:
This is an apparently deliberate deviation of `daf.persistence.Policy` from `dict`-like behavior. 😞 

Since it's undocumented behavior, I guess I could try to handle *both* `None` return and `KeyError`, though that would involve duplicating code.

Reply:
I see. I lean toward whatever will make error messages less cryptic to users, which may mean duplicating code in this case.